tools/wptrunner/wptrunner/testloader.py

```diff
@@ -383,7 +383,7 @@ def group_metadata(cls, state):
     def group(self):
         if not self.current_group or len(self.current_group) == 0:
             try:
-                self.current_group, self.current_metadata = self.test_queue.get(block=False)
+                self.current_group, self.current_metadata = self.test_queue.get(block=True)
```

Comment:
I'm quite worried by this blocking forever; it seems to create a lot of potential for a bug to cause a hang. I'm not sure why reading a populated queue without blocking should be failing, but at the least there should be some kind of timeout.

Reply:
Hi James, I updated per your suggestion. Can you pls review again? Thanks!